ui/src/Update.elm

```diff
@@ -21,29 +21,43 @@ navigateToPage ({ filters } as model) newPage =
             ! [ newUrl <| urlFromRoute updatedRoute ]
 
 
-updateFilters : NewFilter -> Filters -> Filters
-updateFilters newFilter filters =
-    case newFilter of
-        ClearAll ->
-            Init.initFilters
+updateFilters : Facets -> NewFilter -> Filters -> Filters
+updateFilters facets newFilter filters =
+    let
+        toggleFilter facets value values =
+            let
+                newValues =
+                    if List.member value values then
+                        List.filter (\v -> v /= value) values
+                    else
+                        value :: values
+            in
+                if List.length newValues == List.length facets then
```

Comment:
I've tried it, and indeed unchecking everything when you check the last filter is confusing. I believe there's two options there:
1/ if everything is checked, just leave it as is
2/ check everything by default, and add a select all/none button

Searchkit uses the first approach.